pkg/virt-handler/vm.go

```diff
@@ -2638,14 +2638,14 @@ func (d *VirtualMachineController) calculateVmPhaseForStatusReason(domain *api.D
 func (d *VirtualMachineController) addFunc(obj interface{}) {
 	key, err := controller.KeyFunc(obj)
 	if err == nil {
-		d.vmiExpectations.LowerExpectations(key, 1, 0)
+		d.vmiExpectations.CreationObserved(key)
 		d.Queue.Add(key)
 	}
 }
 func (d *VirtualMachineController) deleteFunc(obj interface{}) {
 	key, err := controller.KeyFunc(obj)
 	if err == nil {
-		d.vmiExpectations.LowerExpectations(key, 1, 0)
+		d.vmiExpectations.ControllerExpectationsInterface.DeletionObserved(key)
```

Comment:
I know this is weird, but we aren't using expectations here to determine if a VMI has been created or deleted, we're using expectations to determine when a change to the VMI has been received by the informer.

So this is technically correct the way was previously. We aren't detecting creations or deletions, just that the VMI changed.

Reply:
Interesting, this wasn't clear, we should have this commented out in the code to make it easier to follow